source/Plugins.cpp

```diff
@@ -221,26 +221,26 @@ const Plugin *Plugins::Load(const filesystem::path &path)
 			aboutText += child.Token(1) + '\n';
 		else if(key == "version" && hasValue)
 			version = child.Token(1);
-		else if(key == "authors" && child.HasChildren())
+		else if(key == "authors")
 			for(const DataNode &grand : child)
 				authors.insert(grand.Token(0));
-		else if(key == "tags" && child.HasChildren())
+		else if(key == "tags")
 			for(const DataNode &grand : child)
 				tags.insert(grand.Token(0));
-		else if(key == "dependencies" && child.HasChildren())
+		else if(key == "dependencies")
 		{
 			for(const DataNode &grand : child)
 			{
 				const string &grandKey = grand.Token(0);
-				if(grandKey == "game version")
+				if(grandKey == "game version" && hasValue)
 					dependencies.gameVersion = grand.Token(1);
```

Comment:
Should we add bounds checking to DataNode::Token? Value, BoolValue, IsNumber, and IsBool all do bounds checking on the tokens list, but Token has a comment explicitly stating that it doesn't bounds check (although with no reason for the lack of bounds checking). That way this would have printed an error instead of crashing.

Reply:
We already have places with size checks for this exact purpose, so if we want to remove all instances rendered unnecessary by that change, it would be a global refactor.

Reply:
I don't see how having Token bounds check, just like various other DataNode functions do (i.e. we already have some redundancy of bounds checking when loading objects), to avoid crashing the game would necessitate removing size checks from Load functions.

Tripping a bounds check in DataNode is a developer error. We'd still want objects to bounds check while loading as to not try and store a bad value in a universe object due to a content creator error.

Reply:
idk I just like to keep the code clean in such cases